dumpling/export/dump.go

```diff
@@ -1573,13 +1576,14 @@ func setSessionParam(d *Dumper) error {
 			conf.ServerInfo.HasTiKV, err = CheckTiDBWithTiKV(pool)
 			if err != nil {
 				d.L().Info("cannot check whether TiDB has TiKV, will apply tidb_snapshot by default. This won't affect dump process", log.ShortError(err))
+				ignorableSnapshot = true
```

Comment:
Here we can't distinguish the snapshot is set by user or dumpling?

Reply:
No we can't.

Reply:
I think for user specified snapshot, we can report the error that this TiDB can't set snapshot of that time. For dumpling's automatic snapshot we can fallback to no snapshot and also change the rebuild connection strategy to non-retryable.

Reply:
I think it a second time. I think changing retry strategy is too complex. If we fail to set snapshot for ConsistencySnapshot, we can continue to ConsistencyNone or ConsistencyLock and notify the users in log. Or, we can directly return an error and give a proper prompt to users to notify them to manually switch consistency to none/lock instead of auto/snapshot. Changing retry strategy may confuse our users and is more difficult to maintain to me. WDYT? @lance6716

Reply:
yes a simple fix like automatically fallback to consistency none, or output a message is ok. Let's ask another reviewer @gozssky